src/Analyzers/VisualBasic/Analyzers/SimplifyObjectCreation/VisualBasicSimplifyObjectCreationDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,64 @@
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.CodeStyle
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic.CodeStyle
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.SimplifyObjectCreation
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend NotInheritable Class VisualBasicSimplifyObjectCreationDiagnosticAnalyzer
+        Inherits AbstractBuiltInCodeStyleDiagnosticAnalyzer
+
+        Public Sub New()
+            MyBase.New(
+                diagnosticId:=IDEDiagnosticIds.SimplifyObjectCreationDiagnosticId,
+                enforceOnBuild:=EnforceOnBuildValues.SimplifyObjectCreation,
+                [option]:=VisualBasicCodeStyleOptions.PreferSimplifiedObjectCreation,
+                language:=LanguageNames.VisualBasic,
+                title:=New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Object_creation_can_be_simplified), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)))
+        End Sub
+
+        Protected Overrides Sub InitializeWorker(context As AnalysisContext)
+            context.RegisterSyntaxNodeAction(AddressOf AnalyzeVariableDeclarator, SyntaxKind.VariableDeclarator)
+        End Sub
+
+        Public Overrides Function GetAnalyzerCategory() As DiagnosticAnalyzerCategory
+            Return DiagnosticAnalyzerCategory.SemanticSpanAnalysis
+        End Function
+
+        Private Sub AnalyzeVariableDeclarator(context As SyntaxNodeAnalysisContext)
+            ' Finds and reports syntax on the form:
+            ' Dim x As SomeType = New SomeType()
+            ' which can be simplified to
+            ' Dim x As New SomeType()
+
+            Dim node = context.Node
+            Dim tree = node.SyntaxTree
+            Dim cancellationToken = context.CancellationToken
+
+            Dim styleOption = context.Options.GetOption(VisualBasicCodeStyleOptions.PreferSimplifiedObjectCreation, tree, cancellationToken)
+            If Not styleOption.Value Then
+                Return
+            End If
+
+            Dim variableDeclarator = DirectCast(node, VariableDeclaratorSyntax)
+            Dim asClauseType = variableDeclarator.AsClause?.Type()
+            If asClauseType Is Nothing Then
+                Return
+            End If
+
+            Dim objectCreation = TryCast(variableDeclarator.Initializer?.Value, ObjectCreationExpressionSyntax)
+            If objectCreation Is Nothing Then
+                Return
+            End If
+
+            Dim symbolInfo = context.SemanticModel.GetTypeInfo(objectCreation, cancellationToken)
+            If symbolInfo.Type IsNot Nothing AndAlso symbolInfo.Type.Equals(symbolInfo.ConvertedType, SymbolEqualityComparer.Default) Then
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, variableDeclarator.GetLocation()))
```

Comment:
i think this loses the value of hte style option.  i.e. the severity.

Reply:
@CyrusNajmabadi Can you elaborate more on this?

Reply:
Could you clarify how that works?  How did the diagnostic know what severity the user has set this particular option to?

Reply:
I think this happens around the following, not sure enough:

https://github.com/dotnet/roslyn/blob/1cca63b5d8ea170f8d8e88e1574aa3ebe354c23b/src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs#L213-L224

https://github.com/dotnet/roslyn/blob/1cca63b5d8ea170f8d8e88e1574aa3ebe354c23b/src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs#L195-L215

Reply:
I do not understand why some of these anlayzers are using this form, and some are using DiagnosticHelper.Create.  Can you please use DiagnosticHelper.Create to be consistent with the rest of the analyzers in this domain.  If that approach is incorrect we cna fix them all up together.

@sharwell @mavasani to advise.  Note: this is one of the reasons i *really* dislike making pattern changes in a piecemeal fashion.  I cannot go and look and see what is the right way to do something because we aren't consistent in our own codebase.

Reply:
@mavasani can you verify that all the above is correct?  That it's fine for an anlyzer to just create the Diagnostic and not pass the severity.  If so, i'm fine with this piece.

Reply:
We have 2 core scenarios here to consider, let me elaborate:
1) Command line builds - This scenario will have no impact on whether we use `DiagnosticHelper.Create(..., Notification.Severity, ...)` or `Diagnostic.Create(...)`, with the latter ignoring `Notification.Severity` from the code style option.
2) IDE background analysis - This scenario will have different user experience based on the chosen approach. Using `DiagnosticHelper.Create(..., Notification.Severity, ...)` would ensure the option severity set in Tools/Option or via the `:severity` syntax in editorconfig will be respected. Using `Diagnostic.Create(...)` would lead to this severity setting being ignored even in the IDE.

Overall it proves the point that we were just discussing at the design meeting - determining the reported severity of diagnostics should not be determined by the analyzers, but instead better for analyzer driver to do the right thing. I think it is fine to leave it in Dev16 and we can hopefully fix the core issue properly in the compiler/analyzer driver layer in Dev17 so users and analyzer authors have the minimal overhead for getting the most intuitive functionality.

Reply:
I see.  This code is incorrect then. @Youssef1313 can you revise this to address this issue?

Reply:
> Overall it proves the point that we were just discussing at the design meeting - determining the reported severity of diagnostics should not be determined by the analyzers, but instead better for analyzer driver to do the right thing.

I'm fine with that.  My general point is that that should still be done in a way taht is palatable to the person who owns the editorconfig file.  :)